src/sql/platform/capabilities/common/capabilitiesServiceImpl.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter } from 'vs/base/common/event';
```

Comment:
This is the biggest change. I rewrote the capabilities to not have a cache and not using a registry. It cleans up the dependency chain a good bit and makes the code cleaner. I didn't see any side effects in my testing. It should work exactly the same.